failure of update github.io to clone checkstyle.github.io

https://github.com/checkstyle/checkstyle/runs/8179173614?check_suite_focus=true
failure of update github.io
```
Clone by ssh only to avoid passwords on push ...
Cloning into 'checkstyle.github.io'...
Warning: Permanently added the ECDSA host key for IP address '140.82.114.3' to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'checkstyle.github.io'...
Warning: Permanently added the ECDSA host key for IP address '140.82.113.3' to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'checkstyle.github.io'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'checkstyle.github.io'...
Warning: Permanently added the ECDSA host key for IP address '140.82.114.4' to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'checkstyle.github.io'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
./.ci/update-github-io.sh: line 23: cd: .ci-temp/checkstyle.github.io: No such file or directory
Error: Process completed with exit code 1.
```

This failing with:
```
remote: Permission to checkstyle/checkstyle.github.io.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/checkstyle/checkstyle.github.io/': The requested URL returned error: 403
Push site content to remote ...
We do force to avoid history changes as github.io shows only HEAD.
Error: Process completed with exit code 128.
```

Pretty reasonable security limit, no push to different repo. We need to use personal token for this provided in security variables.

--- a/.ci/generate-website.sh
+++ b/.ci/generate-website.sh
@@ -6,7 +6,7 @@ source ./.ci/util.sh
 
 if [[ -z $1 ]]; then
   echo "version is not set"
-  echo "Usage: update-github-io.sh <version>"
+  echo "Usage: generate-website.sh <version>"
   exit 1
 fi
 
@@ -18,8 +18,4 @@ git checkout "checkstyle-$TARGET_VERSION"
 echo "Generating web site"
 mvn -e --no-transfer-progress site -Pno-validations -Dmaven.javadoc.skip=false
 
-echo "Clone by ssh only to avoid passwords on push ..."
-checkout_from git@github.com:checkstyle/checkstyle.github.io.git
-cd .ci-temp/checkstyle.github.io
-git rm -rf *
-git checkout HEAD -- CNAME
+git checkout origin/master

--- a/.ci/push-website-to-github-io.sh
+++ b/.ci/push-website-to-github-io.sh
@@ -1,0 +1,25 @@
+#!/bin/bash
+set -e
+
+source ./.ci/util.sh
+
+if [[ -z $1 ]]; then
+  echo "version is not set"
+  echo "Usage: push-website-to-github-io.sh <version>"
+  exit 1
+fi
+
+TARGET_VERSION=$1
+echo TARGET_VERSION="$TARGET_VERSION"
+
+cd .ci-temp/checkstyle.github.io/
+git config user.name 'github-actions[bot]'
+git config user.email 'github-actions[bot]@users.noreply.github.com'
+git rm -rf *
+git checkout HEAD -- CNAME
+cp -R ../../target/site/* .
+git add .
+git commit -m "release $TARGET_VERSION"
+echo "Push site content to remote ..."
+echo "We do force to avoid history changes, we do not need history as github.io shows only HEAD."
+git push origin --force

--- a/.github/workflows/update_github_io.yml
+++ b/.github/workflows/update_github_io.yml
@@ -17,16 +17,17 @@ jobs:
       - name: Checkout the latest code
         uses: actions/checkout@v3
         with:
+          repository: checkstyle/checkstyle
           fetch-depth: 0
       - name: Run Shell Script
         run: |
-          ./.ci/update-github-io.sh ${{ github.event.inputs.version }}
-      - name: Commit and Push
+          ./.ci/generate-website.sh ${{ github.event.inputs.version }}
+      - name: Checkout checkstyle.github.io repo
+        uses: actions/checkout@v3
+        with:
+          repository: checkstyle/checkstyle.github.io
+          token: ${{ secrets.PAT }}
+          path: .ci-temp/checkstyle.github.io/
+      - name: Commit and Push to checkstyle.github.io repo
         run: |
-          cd .ci-temp/checkstyle.github.io
-          git config user.name 'github-actions[bot]'
-          git config user.email 'github-actions[bot]@users.noreply.github.com'
-          git add . && git commit -m "release ${{ github.event.inputs.version }}"
-          echo "Push site content to remote ..." 
-          echo "We do force to avoid history changes as github.io shows only HEAD." 
-          git push origin --force 
+          ./.ci/push-website-to-github-io.sh ${{ github.event.inputs.version }}

